### DISCUSSION TITLE
`no applicable key found in the JSON Web Key Set` when trying to `jwtVerify` with JWS

### DISCUSSION THREAD

**Discussion by TimC0de** [panva/jose#475] [Q&A] [answered]
Here is the actual code

```
const { payload } = await jose.jwtVerify(token, jose.createLocalJWKSet({keys: [
        {
            "alg": "RS256",
            "e": "AQAB",
            "key_ops": [],
            "kid": "<some_kid>",
            "kty": "RSA",
            "n": "<some_key>",
            "use": "sig",
            "x5c": [
                "<something_else>"
            ],
            "x5t": "<more_of_something>",
            "KeySize": 4096,
            "HasPrivateKey": false
        }
    ]}))
```

and for some reason I get error with message `no applicable key found in the JSON Web Key Set`
I will appreciate any help

**Comment by panva** [comment, marked as answer]
Empty key ops.

**Reply by TimC0de** [reply]
> Empty key ops.

Okay, I fixed the problem with key_ops and now I'm getting error message `JWSInvalid: JWS Protected Header is invalid`
Any ideas about it?

**Reply by panva** [reply]
It's pretty self explanatory, isn't it? I can hardly have any more ideas since I don't know the token you're verifying.

**Comment by ic-agouveia** [comment]
If you are using auth0, this is triggered if the access token you are using from the _Auth0 Management API_ (**Test** tab) does not relate to the code/API you are testing (e.g. of a different tenant).

Just ensure the tenant you copied that token from is the one you are actually testing.
